### PR TITLE
fix: site backups not working in stg #963

### PR DESCRIPTION
**Pull request by ClaireMao1996**
Close https://github.com/samatechtw/pubstudio/issues/963

### CODE REVIEW

File: backend/lib-shared-site-api/src/clients/s3_client.rs [hunk 1]
```diff
@@ -118,6 +118,25 @@ impl S3Client {
         )
     }
 
+    pub fn presign_put_site_backup(
```
Comment (sampullman) [inline backend/lib-shared-site-api/src/clients/s3_client.rs:121]: Can we reuse the existing `presign_put` function? It seems like the only difference is we need to pass `self.backup_bucket` as a parameter.

**Comment by sampullman** [comment]
Thanks! I just included one comment.

I think it's possible to just call `PutObject` directly without presign, but it's fine to reuse the existing presign method since it's already well tested.

**Comment by ClaireMao1996** [comment]
I forgot to comment out the dev environment, so all my tests were successful locally previously😅. I just replaced the `s3_secret_access_key` with the one from the staging for testing. I found that using presign doesn't work; the `MissingContentLength` error still occurs. When I switch back to directly calling the PutObject method and additionally add the `header(header::CONTENT_LENGTH, content_size.to_string())`, another error appears: `connection error: unexpected end of file`

截圖 2024-01-23 下午1 48 05

**Comment by sampullman** [comment]
Maybe the content_size calculation doesn't match the real file size? I think that could cause both the errors you're seeing.

**Comment by ClaireMao1996** [comment]
The size calculation is to obtain the size of the .db file after performing VACUUM, I'll double check for that!

**Comment by sampullman** [comment]
The changes make sense to me! Do you think it's ready to merge?

**Comment by ClaireMao1996** [comment]
I think we can merge and see how it goes. I’ll keep tracing the logs tomorrow! Thanks🙏

**Pull request merged**